portal-api/src/v1/gef/controllers/validation/supportingInfo.js

```diff
@@ -1,31 +1,26 @@
 const { STATUS } = require('../../enums');
 
-/* eslint-disable consistent-return */
-const hasRequiredItems = (doc) => {
-  const required = [];
+const supportingInfoStatus = (supportingInfo) => {
+  if (supportingInfo) {
+    const requiredCount = supportingInfo.requiredFields?.length;
 
-  if (!doc?.manualInclusion?.length) {
-    required.push('manualInclusion');
-  }
+    const supportingInfoAnswers = supportingInfo;
+    delete supportingInfoAnswers.requiredFields;
+    delete supportingInfoAnswers.status;
 
-  if (!doc?.securityDetails?.exporter || !doc?.securityDetails?.application) {
-    required.push('securityDetails');
-  }
+    const answeredCount = Object.keys(supportingInfoAnswers).length;
 
-  return required;
-};
+    if (!answeredCount) {
+      return STATUS.NOT_STARTED;
+    }
+
+    if (answeredCount === requiredCount) {
+      return STATUS.COMPLETED;
+    }
 
-const supportingInfoStatus = (doc) => {
-  const requiredCount = hasRequiredItems(doc).length;
-  if (requiredCount === 2) {
-    return STATUS.NOT_STARTED;
-  }
-  if (requiredCount > 0) {
     return STATUS.IN_PROGRESS;
   }
-  if (requiredCount === 0) {
-    return STATUS.COMPLETED;
-  }
+  return STATUS.NOT_STARTED;
```

Comment:
Do we have a test for this file - I think the whole validation directory has poor test coverage, good to add as we go along

Reply:
yup - see above in this file
`/api-tests/v1/gef/validation/supportingInfo.api-test.js `